### PR TITLE
Fab 323 charts help text

### PR DESCRIPTION
**Pull request by nikodemadamski**
added all of the data visualisation helptext

### CODE REVIEW

**Review by D22124491-NaveenMaheswaran — APPROVED** [review]

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 1]
```diff
@@ -11,29 +11,89 @@
...
-    "NUMBEROFFIELDS":{
-        "WHAT":"Displays the count of fields in the dataset.",
-        "WHY":""
+    "bubble-chart":{
+        "WHAT":"Visualises the categorical features within a column as bubbles.",
+        "WHY":"Use to identify categorical distribution. Large bubbles may indicate asimetric data points needing normalisation."
         
         
     },
-    "NUMBEROFFIELDS":{
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:26]: Good to remove chart is self explanatory

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 2]
```diff
@@ -11,29 +11,89 @@
...
+        "WHY":"Helpful for understanding when to collapse rare categories or when binning might be necessary for numeric to categorical conversions."
         
         
     },
-    "NUMBEROFFIELDS":{
-        "WHAT":"Displays the count of fields in the dataset.",
-        "WHY":""
+    "correlation-scatter-plot":{
+        "WHAT":"Illustrates the relationship between two variables and their correlation coefficient.",
+        "WHY":"Use to determine if fields are redundant (high correlation) or if PCA might be useful for dimensionality reduction."
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:34]: Would remove the line about PCA possibly as we dont have that in the app anymore

Comment (SeanMcCrossanTUD) [reply Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:34]: Or keep it, not sure will let you decide, users will have to apply PCA out of the app

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 3]
```diff
@@ -11,29 +11,89 @@
...
+ 
+    },
+    "data-types-chart":{
+        "WHAT":"Categorizes data into different types (numeric, categorical, etc.).",
+        "WHY":"Allows for a better fundemental understanding of data type distribution"
+        
+ 
+    },
+    "duplicate-chart":{
+        "WHAT":"Visualizes the frequency of duplicate records in a dataset.",
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:51]: Removed the merged part as we dont do that, just remove

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 4]
```diff
@@ -11,29 +11,89 @@
...
+        "WHAT":"Visualises the categorical features within a column as bubbles.",
+        "WHY":"Use to identify categorical distribution. Large bubbles may indicate asimetric data points needing normalisation."
         
         
     },
-    "NUMBEROFFIELDS":{
-        "WHAT":"Displays the count of fields in the dataset.",
-        "WHY":""
+    "cardinality":{
+        "WHAT":"Shows the uniqueness of data by representing the number of distinct values in a column.",
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:27]: High cardinality can introduce the curse of dimensionailty, columns may need or encoded if using them as a predictor variable in ML

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 5]
```diff
@@ -11,29 +11,89 @@
...
+ 
+    },
+    "missing-values-chart":{
+        "WHAT":"Highlights the number of missing values within a column in a dataset.",
+        "WHY":"Directs where to fill missing values or to decide if a column with too many missing values should be removed."
+        
+ 
+    },
+    "outliers-scatter-plot":{
+        "WHAT":"Identifies outlier data points in a dataset.",
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:69]: Could you mention outliers identified using Z score and a threshold of 3 standard deviation points.

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 6]
```diff
@@ -11,29 +11,89 @@
...
         
+ 
+    },
+    "uniqueness-metric":{
+        "WHAT":"Displays the level of unique data entries, highlighting potential redundancy in the dataset.",
+        "HOW TO IMPOROVE":"Apply 'Collapse Rare Categories' to reduce overfitting in machine learning models and simplify analysis."
+        
+ 
+    },
+    "readability-metric":{
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:86]: Really good

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 7]
```diff
@@ -11,29 +11,89 @@
...
+    },
+    "outliers-scatter-plot":{
+        "WHAT":"Identifies outlier data points in a dataset.",
+        "WHY":"Pinpoints outliers for removal or further examination."
+        
+ 
+    },
+    "completeness-metric":{
+        "WHAT":"Measures the proportion of non-missing entries in the dataset to assess data completeness.",
+        "HOW TO IMPOROVE":"to improve the score, look into the 'Fill Missing Values' function."
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:76]: Like this

File: Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json [hunk 8]
```diff
@@ -11,29 +11,89 @@
...
         
+ 
+    },
+    "uniqueness-metric":{
```
Comment (SeanMcCrossanTUD) [inline Frontend-Angular/data-polish-frontend/src/assets/Dataprofile_config.json:80]: Would mention curse of dimensionality and why that impacts but overall like this

**Review by SeanMcCrossanTUD — COMMENTED** [review]
Overall like this but have some suggested changes which you can agree or disagree with.

**Pull request merged**